sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -20,18 +20,24 @@
 #include <vector>
 
 #include "usm_allocator.hpp"
+#include <iostream>
 
 namespace settings {
-// Size of the slab which is going to be requested from the system.
-static constexpr size_t SlabSize = 64 * 1024; // 64K
+// Minimum allocation size that will be requested from the system.
+static constexpr size_t SlabSize = 64 * 1024; // 64KB
+
+// Allocations <= ChunkCutOff will use chunks from individual slabs.
+// Allocations >  ChunkCutOff will be rounded up to a multiple of
+// SlabSize and allocated to occupy the whole slab.
+static constexpr size_t ChunkCutOff = SlabSize / 2;
 // The largest size which is allocated via the allocator.
 // Allocations with size > CutOff bypass the USM allocator and
 // go directly to the runtime.
-static constexpr size_t CutOff = SlabSize / 2;
+static constexpr size_t CutOff = (size_t)1 << 31; // 2GB
```

Comment:
OK, I'll make it small. This might have to be changed later based on experience with apps, but making it small now reduces chance of running out of memory. The setting of MaxPoolSize limits total memory for pool:
SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR=[<MaxPoolableSize>][,[<Capacity>][,[<MaxPoolSize>]]]

Reply:
SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR_SETTINGS=\[<MaxPoolableSize>\]\[\[<Capacity>\]\[,\[<MaxPoolSize>\]\]\]

Reply:
Trying again: "
SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR_SETTINGS=[<MaxPoolableSize>][,[<Capacity>][,[<MaxPoolSize>]]]"

Reply:
SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR_SETTINGS=MaxPoolableSize,Capacity,MaxPoolSize